src/asm/main.cpp

```diff
@@ -376,10 +377,8 @@ int main(int argc, char *argv[])
 		}
 	}
 
-	if (targetFileName == NULL)
-		targetFileName = objectName;
-	else
-		targetFileName[targetFileNameLen - 1] = '\0'; // Overwrite the last space
```

Comment:
If this is removed, then where is `targetFileName` terminated?

Reply:
`memcpy(&targetFileName[targetFileNameLen], newTarget, newTargetLen)` does the termination, since `newTargetLen` is `strlen(newTarget) + 1`.

Reply:
Isn't that copying the terminator a bunch of times and overwriting it each?

Reply:
There's no redundant work, if that's what you're asking. Before this PR, every terminator would be overwritten by a space after being memcpyed, and then this piece of code was necessary to put back the terminator. After this PR, the memcpyed terminator is correctly kept, and the previous terminator is overwritten with a space. (Which does add asymmetry between the first `MT` and all subsequent ones.)

Reply:
That sounds like a bunch of unnecessary work... but OTOH this is seldom run, so I don't feel like nitpicking it here and now.